### ISSUE TITLE
Segmentation fault in single threaded with beagle input

### ISSUE THREAD

**Issue by GenisGE** [GenisGE/evalAdmix#3]
Hi all,

I am not sure if this is a related issue, but I have been getting segmentation faults with even relatively small datasets (60 samples, ~70000 sites), always occurring immediately after finishing the steps `Estimating freqs without ind ##`. I've found this happens on both 0.95 and 0.96, so it may be unrelated to this issue (I haven't yet been able to test on another system, the closest I've come is testing within and outside a singularity container).

The run of evalAdmix looks like this:
```
evalAdmix version 0.96
evalAdmix -beagle results/pargus/beagle/test.beagle.gz -fname results/pargus/ngsadmix/test_K3.fopt.gz -qname results/pargus/ngsadmix/test_K3.qopt -o test.corres 
Going to read beagle file
Input file has dim: nsites=77034 nind=61
                ->K=3   nSites=77034    nInd=61
opening : results/pargus/ngsadmix/test_K3.qopt with x=61 y=3
opening : results/pargus/ngsadmix/test_K3.fopt.gz with x=77034 y=3
Finished calculating normal residuals
[1]    29836 segmentation fault (core dumped) evalAdmix -beagle  -fname  -qname  -o 
```

With the coredump looking like this:
```
gdb -q evalAdmix -c core.29836                            
Reading symbols from evalAdmix...(no debugging symbols found)...done.
BFD: Warning: core.29836 is truncated: expected core file size >= 207376384, found: 12288.
[New LWP 29836]
Failed to read a valid object file image from memory.
Core was generated by `evalAdmix -beagle results/pargus/'.
Program terminated with signal 11, Segmentation fault.
#0  0x0000000000412083 in ngsevalAdmix(double*, int, int, double**, double*, double**, int, double**, double**, int, int, int, char**, int) ()
```
The core dump seems to be missing the whole command when it says `Core was generated by`, but I am rather new at looking at core dumps, so can't say if that is normal or not. Visually inspecting the input files, all seem to be complete and as expected, and this happens on multiple independently created beagle files for my dataset. Would any of you have suggestions given the fault information I get?

Thanks!
Zach

_Originally posted by @zjnolen in https://github.com/GenisGE/evalAdmix/issues/2#issuecomment-1122318615_

**Comment by GenisGE** [comment]
This should be fixed in version 0.961

**Comment by zjnolen** [comment]
Hi Genis,

Pulled the new version and it works on my end now too, thanks for the swift fix and for this great tool!

Best,
Zach
